### ISSUE TITLE
Provide better UX for Clients when idle-timeout is reached

### ISSUE THREAD

**Issue by TomAugspurger** [dask/distributed#4617]
Currently, users who've connected to a scheduler that then times out don't get much information.

```python
In [1]: from distributed import Client

In [2]: client = Client("172.25.136.49:8786")  # started with `dask-scheduler --idle-timeout=10s

In [3]: client
Out[3]: <Client: 'tcp://172.25.136.49:8786' processes=0 threads=0, memory=0 B>
# scheduler times out here <-----------------------------------
In [4]: distributed.client - ERROR - Failed to reconnect to scheduler after 10.00 seconds, closing client
_GatheringFuture exception was never retrieved
future: <_GatheringFuture finished exception=CancelledError()>
asyncio.exceptions.CancelledError
In [4]:
```

I wonder how we can make this UX a bit nicer. It's a bit difficult since the scheduler is the one going away. Tossing some ideas out:

1. scheduler notifies the client that it's timing out, Update the client's repr to include the fact that it's timed out.
2. scheduler notifies the client that it's timing out, Use whatever mechanism is printing that `_GatheringFuture exception` to immediately print that the scheduler has timed out.

cc @jacobtomlinson if you have thoughts on this.

**Comment by jrbourbeau** [comment]
+1 to improving the message that's raised to users here. We might want to raise an informative warning to users when the scheduler times out as updating the client repr might go unnoticed and leave users in a confused state.

**Comment by jrbourbeau** [comment]
Something along the lines of using `Scheduler.report`

https://github.com/dask/distributed/blob/42894418a7c63499eeb240a8cbd3c7cc548bea11/distributed/scheduler.py#L4658

to send a `"idle-timeout"` message to all connected client when the scheduler closes itself due to being idle

https://github.com/dask/distributed/blob/42894418a7c63499eeb240a8cbd3c7cc548bea11/distributed/scheduler.py#L6601-L6606

and then adding a new handler on the clients to process the `"idle-timeout"` message

https://github.com/dask/distributed/blob/42894418a7c63499eeb240a8cbd3c7cc548bea11/distributed/client.py#L717-L725

seems like a reasonable way to handle things

**Comment by jacobtomlinson** [comment]
I totally agree that it would be nice for the scheduler to notify connected clients when timing out.

This also possibly highlights the use case discussed in #3265 where the user wants the `scheduler_timeout` rather than the idle timeout. I don't think `scheduler_timeout` is currently exposed via the CLI, but with that timeout a scheduler with a connected client is not considered idle. Only when all clients have disconnected is the timeout countdown started.
